### PR TITLE
Add game speed dropdown to the mission browser.

### PR DESCRIPTION
**Pull request by pchote**
screen shot 2015-09-19 at 20 37 27

This exposes a game speed dropdown in the mission browser, and makes the difficulty dropdown visible for all mods.  Also renames Default &rarr; Normal speed to be consistent with the difficulties.

### CODE REVIEW

File: OpenRA.Mods.Common/Widgets/Logic/MissionBrowserLogic.cs [hunk 1]
```diff
@@ -186,27 +189,58 @@ void SelectMap(Map map)
 			descriptionPanel.ScrollToTop();
 			descriptionPanel.Layout.AdjustChildren();
 
-			difficultyButton.IsVisible = () => map.Options.Difficulties.Any();
-			if (!map.Options.Difficulties.Any())
-				return;
```
Comment (reaperrr) [inline OpenRA.Mods.Common/Widgets/Logic/MissionBrowserLogic.cs]: > OpenRA.Mods.Common/Widgets/Logic/MissionBrowserLogic.cs:L192: [SA1507] The code must not contain multiple blank lines in a row.

**Comment by pchote** [comment]
Updated.

**Comment by clemty** [comment]
looks good! But why does the player need to set the speed before mission start? Original TD had the speed setting on the same ingame options screen as the volume.

**Comment by GraionDilach** [comment]
A) In TS and RA2, triggers could malfunction on nondefault game speeds which is why they had that slider disabled during a campaign.
B) IIRC, the dynamic changing of game speed settings are still not supported.

**Comment by Mailaender** [comment]
:+1: Works as promised. I see two UI polish issues, but I don't want to derail the discussion here.

**Comment by pchote** [comment]
What are the polish issues?

**Comment by Mailaender** [comment]
https://github.com/OpenRA/OpenRA/issues/9392 and https://github.com/OpenRA/OpenRA/issues/9393 but you don't need to address them here.

**Comment by pchote** [comment]
I think I should fix #9393 here first.

**Comment by pchote** [comment]
Updated.  I have trimmed away the starfield at the top and bottom of the d2k mission previews so that we have enough room to display the mission description.  This conflicts with #9402.

**Comment by pchote** [comment]
Updated again to fix the fmv issue.

I've also reduced the width of the panel, and have fixed the aspect ratio of the FMVs:

screen shot 2015-09-27 at 13 58 32

screen shot 2015-09-27 at 13 58 21

**Comment by obrakmann** [comment]
Looks fine :+1:

**Comment by obrakmann** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/e6de1eee8d83de1e12b697f486e393bad1fc004b)

**Comment by AoAGeneral** [comment]
Tested this in the current bleed version. A nice little addition but im not sure if this is supposed to happen or not.

All the units and buildings have increased time speeds. IE: normal gameplay weapon factories build at 48 seconds. Fastest they build in 24 seconds. This means buggies build in nearly 6 seconds. And since planes fly faster as well they get dropped off nearly instantly which can easily overpower GDI.

Again im not sure if its supposed to speed up production speeds or if its supposed to just speed up the game itself.

**Comment by pchote** [comment]
The game speed is a global modifier.  *Everything* runs faster or slower by the same factor.

**Comment by AoAGeneral** [comment]
Aah I gotcha!

**Pull request merged**